src/target/riscv/riscv-013.c

```diff
@@ -2452,7 +2452,10 @@ static int deassert_reset(struct target *target)
 				return ERROR_FAIL;
 			}
 		}
-		target->state = TARGET_HALTED;
+		if (target->reset_halt)
+			target->state = TARGET_HALTED;
+		else
+			target->state = TARGET_RUNNING;
 		target->debug_reason = DBG_REASON_DBGRQ;
```

Comment:
I don't think any code should look at `target->debug_reason` when `target->state == TARGET_RUNNING`. But for clarity & robustness, we better keep those two aligned:

```suggestion
		if (target->reset_halt) {
			target->state = TARGET_HALTED;
			target->debug_reason = DBG_REASON_DBGRQ;
		} else {
			target->state = TARGET_RUNNING;
			target->debug_reason = DBG_REASON_NOTHALTED;
		}
```